mmv1/third_party/terraform/fwtransport/framework_config.go.erb

```diff
@@ -59,9 +59,9 @@ var defaultClientScopes = []string{
 
 // LoadAndValidateFramework handles the bulk of configuring the provider
 // it is pulled out so that we can manually call this from our testing provider as well
-func (p *FrameworkProviderConfig) LoadAndValidateFramework(ctx context.Context, data fwmodels.ProviderModel, tfVersion string, diags *diag.Diagnostics, providerversion string) {
+func (p *FrameworkProviderConfig) LoadAndValidateFramework(ctx context.Context, data *fwmodels.ProviderModel, tfVersion string, diags *diag.Diagnostics, providerversion string) {
```

Comment:
This PR isn't ready for full review yet, but flagging this change.

Currently the data about the provider's configuration in the config (ie provider block) is received by the provider code and used to populate the `data *fwmodels.ProviderModel` struct [in this code](https://github.com/hashicorp/magic-modules/blob/339a53ce1c9e86acfd20298a522091e982b08cb5/mmv1/third_party/terraform/fwprovider/framework_provider.go.erb#L228-L231). The data struct is only used by passing it into `LoadAndValidateFramework`, and isn't used elsewhere. Because it's only got this one use, I think this change doesn't have much risk associated with it.

Making the function take a pointer instead of a copied value allows the test to see how the data model struct is mutated inside this function.